merkle_patricia_tree/merkle_tree/src/main.rs

```diff
@@ -0,0 +1,533 @@
+                )
+            };
+            
+        });
+    }
+}
+
+fn mutate(input:  &mut [u8]) -> Vec<Vec<u8>> {
+    /// List of mutation strategies which do not require an input database
+    const STRATEGIES: &[for<'a> fn(&'a mut [u8]) -> Vec<u8>; 9] = &[
+        shrink,
+        expand,
+        add_sub,
+        set,
+        swap,
+        copy,
+        inter_splice,
+        insert_rand,
+        overwrite_rand,
+        // Mutator::byte_repeat_overwrite,
+        // Mutator::byte_repeat_insert,
+        // Mutator::magic_overwrite,
+        // Mutator::magic_insert,
+        // Mutator::random_overwrite,
+        // Mutator::random_insert,
+        // Mutator::bit,
+        // Mutator::inc_byte,
+        // Mutator::dec_byte,
+        // Mutator::neg_byte,
+    ];
+
+    let mut vec = vec![];
+
+    vec.push(input.to_vec());
+
+    let mut rng = StdRng::seed_from_u64(1234567);
+
+    for _ in 0..6 {
+        // Pick a random mutation strategy
+        let sel = rng.gen_range(0..STRATEGIES.len() - 1);
+            
+        // Get the strategy
+        let strat = STRATEGIES[sel];
+
+        // Run the mutation strategy
+        let new_input = strat(input);
+
+        // Add new mutation to vec
+        vec.push(new_input);
+    };
+     
+    vec
+}
+
+fn rand_exp(min: usize, max: usize) -> usize {
+
+    if rand_number_in_range(0,1) == 0 {
+        // Half the time, provide uniform
+        rand_number_in_range(min,max)
+    } else {
+        // Pick an exponentially difficult random number
+        let x = rand_number_in_range(min,max);
+        rand_number_in_range(min,x)
+    }
+}
+
+fn rand_offset(input: &[u8]) -> usize {
+
+    let offset_range = input.len() - 1 as usize;
+
+    if offset_range > 1{
+        // just return a random index
+        rand_number_in_range(0,offset_range)
+    } else {
+        // Input is entirely empty, just return index 0 such that
+        // things that insert into the input know that they should
+        // just insert at 0.
+        0
+    }
+}
+
+// generates a random number within a range
+fn rand_number_in_range(min: usize, max: usize) -> usize {
+
+    let mut rng = StdRng::seed_from_u64(1234567);
+    
+    if min == max {
+        // if there´s no range, return the same number
+        min
+    } else { 
+        let random = rng.gen_range(min..max);
+        random
+    }
+}
+
+/// Randomly delete a chunk of the input
+fn shrink(input:  &mut [u8]) -> Vec<u8> {
+    if input.is_empty() {
+        return input.to_vec();
+    }
+  
+    // Pick a random offset to remove data at
+    let offset = rand_offset(input);
+    // Compute the number of bytes we could remove from this offset
+    let can_remove = input.len() - offset;
+
+    // Compute a maximum number of bytes to remove
+    let max_remove = if rand_number_in_range(0,15) != 0 {
+        // 15 in 16 chance of removing at most 16 bytes, this limits the
+        // amount we remove in the most common case
```

Comment:
`rand_number_in_range(0,15)` has a 1 in 15 chance of returning 0, as it doesn't include 15 in the range (i.e. it goes from 0 to 14, for a total of 15 numbers to choose from).

Reply:
fixed the comment